users.c

```diff
@@ -0,0 +1,104 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+
+enum Credential { Author, Customer, Librarian };
+
+enum UserStatus { inactive, active, delinquent };
+
+typedef struct User
+{
+	int user_id;
+	char* full_name;
+	char* login_name;
+	char* password;
+	char* email;
+	enum Credential title;
+	enum UserStatus status;
+	struct User* next;
+} User;
+
+
+User* user_list_head = NULL;
+
+void insertNewUser(int user_id, char* full_name, char* login_name, char* password, char* email, enum Credential title, enum UserStatus status, User** head)
+{
+	User* new_user = malloc(sizeof(User));
+	if (!new_user)
+	{
+		printf("Fail to allocate memory.\n");
+		return;
+	}
+
+	new_user->user_id = user_id;
+	new_user->full_name = _strdup(full_name);
+	new_user->login_name = _strdup(login_name);
+	new_user->password = _strdup(password); // Need to implement encryption
+	new_user->email = _strdup(email);
+	new_user->title = title;
+	new_user->status = status;
+	new_user->next = NULL; // Add new user to end of list
+
+	if (*head)
+	{
+		User* current = *head;
+		while (current->next)
+		{
+			current = current->next;
+		}
+		current->next = new_user;
+	}
+	else
+	{
+		*head = new_user;
+	}
+}
+
+void printUserList(User* head)
+{
+	User* current = head;
+	int count = 1;
+
+	while (current)
+	{
+		printf("User %i\n", count);
+		printf("User ID: %i\n", current->user_id);
+		printf("Full name: %s\n", current->full_name);
+		printf("Login name: %s\n", current->login_name);
+		printf("Password: %s\n", current->password);
+		printf("Email: %s\n", current->email);
+		printf("Title: ");
+		switch (current->title)
+		{
+		case Author:
+			printf("Author\n");
+			break;
+		case Customer:
+			printf("Customer\n");
+			break;
+		case Librarian:
+			printf("Librarian\n");
+			break;
+		default:
+			break;
+		}
+		printf("Status: ");
+		switch (current->status)
+		{
+		case inactive:
```

Comment:
Do some research can we find out field exact names through enums instead of 0, 1, 2, 3